src/ucp/core/ucp_worker.c

```diff
@@ -621,7 +621,10 @@ static ucs_status_t ucp_worker_iface_check_events_do(ucp_worker_iface_t *wiface,
     } else if (*progress_count == 0) {
         /* Arm the interface to wait for next event */
         ucs_assert(wiface->attr.cap.event_flags & UCT_IFACE_FLAG_EVENT_RECV);
-        status = uct_iface_event_arm(wiface->iface, UCT_EVENT_RECV);
+        ucs_assert(wiface->attr.cap.event_flags &
+                   UCT_IFACE_FLAG_EVENT_SEND_COMP);
```

Comment:
maybe use a single assert**v** including both flags?

Reply:
Updated assertv, along with check for iface flags.